File: src/main/java/eu/zavadil/wn/data/tag/Tag.java
package eu.zavadil.wn.data.tag;

import eu.zavadil.wn.data.language.Language;
import jakarta.persistence.Entity;
import jakarta.persistence.ManyToOne;
import lombok.Data;
import lombok.EqualsAndHashCode;

@EqualsAndHashCode(callSuper = true)
@Data
@Entity
public class Tag extends TagBase {

	@ManyToOne
	private Tag synonymOf;

	@ManyToOne
	private Language language;

}

File: src/ui/src/types/Tag.ts
import {EntityWithName} from "zavadil-ts-common";
import {Language} from "./Language";

export type TagBase = EntityWithName & {
	articleCount: number;
}

export type TagStub = TagBase & {
	synonymOfId?: number | null;
	languageId?: number | null;
}

export type Tag = TagBase & {
	synonymOf?: Tag;
	language?: Language;
}
